src/components/ifxTabs/ifxTabs.tsx

```diff
@@ -0,0 +1,70 @@
+//ifxTabs.tsx
+import { Component, h, Prop, State, Element } from '@stencil/core';
+
+export type TabOrientation = 'horizontal' | 'vertical';
+
+
+@Component({
+  tag: 'ifx-tabs',
+  styleUrl: 'ifxTabs.scss',
+  shadow: true
+})
+
+export class IfxTabs {
+  @Element() el: HTMLElement;
+  @Prop() tabs: string[] = [];
+  @Prop() orientation: TabOrientation = 'horizontal';
+  @Prop() small: boolean = false;
+  @State() activeTabIndex: number = 0;
+
+  tabRefs: HTMLElement[] = [];
+
+  setActiveTab(index: number) {
+    this.activeTabIndex = index;
+    const borderElement = this.el.shadowRoot.querySelector('.active-border') as HTMLElement;
+
+    if (borderElement) {
+      if (this.orientation === 'horizontal') {
+        borderElement.style.left = `${this.tabRefs[index].offsetLeft}px`;
+        borderElement.style.width = `${this.tabRefs[index].offsetWidth}px`;
+        borderElement.style.top = '';
+        borderElement.style.height = '';
+      } else {
+        borderElement.style.top = `${this.tabRefs[index].offsetTop}px`;
+        borderElement.style.height = `${this.tabRefs[index].offsetHeight}px`;
+        borderElement.style.left = '';
+        borderElement.style.width = '';
+      }
+    }
+  }
```

Comment:
Why not replace the code inside the if condition with a `State` that gets updated, which then conditionally adds a class on the element, and these styles are all defined in the `scss file`? That would save a lot of code, make it consistent and easier to read and edit.